egui_demo_lib/src/apps/demo/drag_and_drop.rs

```diff
@@ -116,18 +133,24 @@ impl super::View for DragAndDropDemo {
 
         let mut source_col_row = None;
         let mut drop_col = None;
-
         ui.columns(self.columns.len(), |uis| {
-            for (col_idx, column) in self.columns.iter().enumerate() {
+            for (col_idx, column) in self.columns.clone().into_iter().enumerate() {
                 let ui = &mut uis[col_idx];
                 let can_accept_what_is_being_dragged = true; // We accept anything being dragged (for now) ¯\_(ツ)_/¯
                 let response = drop_target(ui, can_accept_what_is_being_dragged, |ui| {
                     ui.set_min_size(vec2(64.0, 100.0));
-
-                    for (row_idx, &item) in column.iter().enumerate() {
-                        let item_id = Id::new("item").with(col_idx).with(row_idx);
+                    for (row_idx, item) in column.iter().enumerate() {
+                        let item_id = Id::new(self.id_source).with(col_idx).with(row_idx);
```

Comment:
Using a single integer as the top-level id source is not idiomatic egui use.

For a demo I think it would suffice with `let id_source = "my_drag_and_drop_demo";` in this function, and then we can remove `Sefl::id_source` and `COUNTER`